editor/src/messages/tool/common_functionality/auto_panning.rs

```diff
@@ -86,3 +86,302 @@ impl AutoPanning {
+
+		let pointer_keys = SelectToolPointerKeys {
+			axis_align: Key::Shift,
+			snap_angle: Key::Control,
+			center: Key::Alt,
+			duplicate: Key::Alt,
+		};
+
+		// Sending multiple outside viewport events to simulate continuous auto-panning
+		for _ in 0..5 {
+			editor.handle_message(SelectToolMessage::PointerOutsideViewport(pointer_keys.clone())).await;
+		}
+
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(2000., 100.),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		// Verifying the rectngle has moved significantly due to auto-panning
+		let final_transform = editor.active_document().metadata().transform_to_viewport(layer);
+		let translation_diff = (final_transform.translation - initial_transform.translation).length();
+
+		assert!(translation_diff > 10., "Rectangle should have moved significantly due to auto-panning (moved by {})", translation_diff);
+	}
+
+	#[tokio::test]
+	async fn test_select_tool_resizing_auto_panning() {
+		let mut editor = EditorTestUtils::create();
+		editor.new_document().await;
+		editor.drag_tool(ToolType::Rectangle, 50., 50., 150., 150., ModifierKeys::empty()).await;
+		let layer = editor.active_document().metadata().all_layers().next().unwrap();
+		let initial_transform = editor.active_document().metadata().transform_to_viewport(layer);
+
+		editor.select_tool(ToolType::Select).await;
+		editor.left_mousedown(150., 150., ModifierKeys::empty()).await; // Click near edge for resize handle
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(150., 150.),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		editor.handle_message(TransformLayerMessage::BeginScale).await;
+
+		// Moving cursor to trigger auto-panning
+		editor.move_mouse(2000., 2000., ModifierKeys::empty(), MouseKeys::LEFT).await;
+
+		let pointer_keys = SelectToolPointerKeys {
+			axis_align: Key::Shift,
+			snap_angle: Key::Control,
+			center: Key::Alt,
+			duplicate: Key::Alt,
+		};
+
+		// Simulatiing auto-panning for several frames
+		for _ in 0..5 {
+			editor.handle_message(SelectToolMessage::PointerOutsideViewport(pointer_keys.clone())).await;
+		}
+
+		editor.handle_message(TransformLayerMessage::ApplyTransformOperation { final_transform: true }).await;
+
+		// Verifying the transform has changed (scale component should be different)
+		let final_transform = editor.active_document().metadata().transform_to_viewport(layer);
+
+		// Comparing transform matrices to detect scale changes
+		let initial_scale = initial_transform.matrix2.determinant().sqrt();
+		let final_scale = final_transform.matrix2.determinant().sqrt();
+		let scale_ratio = final_scale / initial_scale;
+
+		assert!(
+			scale_ratio > 1.1 || scale_ratio < 0.9,
```

Comment:
Why can you not determine what happens to the scale? Surely the scale should increase.